src/middlewared/middlewared/plugins/virt/instance.py

```diff
@@ -98,10 +98,22 @@ async def query(self, filters, options):
 
     @private
     async def validate(self, new, schema_name, verrors, old=None):
+        # Do not validate image_choices because its an expansive operation, just fail on creation
+
         if not old and await self.query([('name', '=', new['name'])]):
             verrors.add(f'{schema_name}.name', f'Name {new["name"]!r} already exists')
 
-        # Do not validate image_choices because its an expansive operation, just fail on creation
+        sysinfo = None
+        if new.get('memory'):
+            sysinfo = await self.middleware.call('system.info')
```

Comment:
Instead of calling the heavy `system.info`, you could call `system.mem_info` and `system.cpu_info`